<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="2" ignored="1" total="4" passed="1">
  <reporter-output>
    <line>
      <![CDATA[actual count: 2]]>
    </line>
  </reporter-output>
  <suite name="Default suite" duration-ms="32637" started-at="2025-04-01T13:40:18 IST" finished-at="2025-04-01T13:40:51 IST">
    <groups>
      <group name="sanity">
        <method signature="InventoryPageTest.verifyProductsLogoTest()[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="verifyProductsLogoTest" class="TestClasses.InventoryPageTest"/>
      </group> <!-- sanity -->
    </groups>
    <test name="Default test" duration-ms="32637" started-at="2025-04-01T13:40:18 IST" finished-at="2025-04-01T13:40:51 IST">
      <class name="TestClasses.InventoryPageTest">
        <test-method status="PASS" signature="setup()[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="setup" is-config="true" duration-ms="13383" started-at="2025-04-01T13:40:18 IST" finished-at="2025-04-01T13:40:31 IST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="FAIL" signature="verifyCartFunctionalityTest()[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="verifyCartFunctionalityTest" duration-ms="126" started-at="2025-04-01T13:40:31 IST" finished-at="2025-04-01T13:40:32 IST">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [https://www.saucedemo.com/cart.html1] but found [https://www.saucedemo.com/cart.html]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [https://www.saucedemo.com/cart.html1] but found [https://www.saucedemo.com/cart.html]
at org.testng.Assert.fail(Assert.java:97)
at org.testng.Assert.assertEqualsImpl(Assert.java:136)
at org.testng.Assert.assertEquals(Assert.java:118)
at org.testng.Assert.assertEquals(Assert.java:575)
at org.testng.Assert.assertEquals(Assert.java:585)
at TestClasses.InventoryPageTest.verifyCartFunctionalityTest(InventoryPageTest.java:60)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:584)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:172)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:804)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:145)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(Unknown Source)
at org.testng.TestRunner.privateRun(TestRunner.java:770)
at org.testng.TestRunner.run(TestRunner.java:591)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:402)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:396)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:355)
at org.testng.SuiteRunner.run(SuiteRunner.java:304)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1180)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1102)
at org.testng.TestNG.runSuites(TestNG.java:1032)
at org.testng.TestNG.run(TestNG.java:1000)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- verifyCartFunctionalityTest -->
        <test-method status="PASS" signature="closeBrowser(org.testng.ITestResult)[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="closeBrowser" is-config="true" duration-ms="353" started-at="2025-04-01T13:40:32 IST" finished-at="2025-04-01T13:40:32 IST">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyCartFunctionalityTest status=FAILURE method=InventoryPageTest.verifyCartFunctionalityTest()[pri:0, instance:TestClasses.InventoryPageTest@1563da5] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeBrowser -->
        <test-method status="PASS" signature="setup()[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="setup" is-config="true" duration-ms="8747" started-at="2025-04-01T13:40:32 IST" finished-at="2025-04-01T13:40:41 IST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="FAIL" signature="verifyProductsLogoTest()[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="verifyProductsLogoTest" duration-ms="76" started-at="2025-04-01T13:40:41 IST" finished-at="2025-04-01T13:40:41 IST">
          <exception class="java.lang.ArithmeticException">
            <message>
              <![CDATA[/ by zero]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.ArithmeticException: / by zero
at TestClasses.InventoryPageTest.verifyProductsLogoTest(InventoryPageTest.java:45)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:584)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:172)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:804)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:145)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(Unknown Source)
at org.testng.TestRunner.privateRun(TestRunner.java:770)
at org.testng.TestRunner.run(TestRunner.java:591)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:402)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:396)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:355)
at org.testng.SuiteRunner.run(SuiteRunner.java:304)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1180)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1102)
at org.testng.TestNG.runSuites(TestNG.java:1032)
at org.testng.TestNG.run(TestNG.java:1000)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.ArithmeticException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- verifyProductsLogoTest -->
        <test-method status="PASS" signature="closeBrowser(org.testng.ITestResult)[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="closeBrowser" is-config="true" duration-ms="462" started-at="2025-04-01T13:40:41 IST" finished-at="2025-04-01T13:40:41 IST">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyProductsLogoTest status=FAILURE method=InventoryPageTest.verifyProductsLogoTest()[pri:0, instance:TestClasses.InventoryPageTest@1563da5] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeBrowser -->
        <test-method status="PASS" signature="setup()[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="setup" is-config="true" duration-ms="8695" started-at="2025-04-01T13:40:41 IST" finished-at="2025-04-01T13:40:50 IST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="PASS" signature="verifySortTest()[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="verifySortTest" duration-ms="579" started-at="2025-04-01T13:40:50 IST" finished-at="2025-04-01T13:40:50 IST">
          <reporter-output>
            <line>
              <![CDATA[actual count: 2]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifySortTest -->
        <test-method status="PASS" signature="closeBrowser(org.testng.ITestResult)[pri:0, instance:TestClasses.InventoryPageTest@1563da5]" name="closeBrowser" is-config="true" duration-ms="151" started-at="2025-04-01T13:40:51 IST" finished-at="2025-04-01T13:40:51 IST">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifySortTest status=SUCCESS method=InventoryPageTest.verifySortTest()[pri:0, instance:TestClasses.InventoryPageTest@1563da5] output=actual count: 2]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeBrowser -->
      </class> <!-- TestClasses.InventoryPageTest -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
